### ISSUE TITLE
Bug: Error or bug using Prisma with DriverAdapter with PostgreSQL database Neon

### ISSUE THREAD

**Issue by EddyerDevv** [prisma/prisma#23903]
### Bug description

I'm using NextJS and NextAuth in its v5 beta version, when I use Prisma with Neon as a postgresql database, and using the adapters, to remove the Edge Functions error, I get that, and it is a basic code, what can be the solution, or is it a bug?

## Error

![image](https://github.com/prisma/prisma/assets/112896919/ed4fd3cb-78b9-4225-9f55-f7ce216a0344)

## Code

![image](https://github.com/prisma/prisma/assets/112896919/e2b2c369-5ee0-4438-b8b4-f727b86d46ef)

### How to reproduce

The error or bug occurs when it is already running, it appears for a few moments or several logs and then does not appear.

### Prisma information

```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["driverAdapters"]
}

model User {
  id            String    @id @default(cuid())
  name          String?
  email         String    @unique
  role          UserRole  @default(USER)
  emailVerified DateTime?
  image         String?
  accounts      Account[]
  sessions      Session[]

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model Account {
  userId            String
  type              String
  provider          String
  providerAccountId String
  refresh_token     String?
  access_token      String?
  expires_at        Int?
  token_type        String?
  scope             String?
  id_token          String?
  session_state     String?

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@id([provider, providerAccountId])
}

model Session {
  sessionToken String   @unique
  userId       String
  expires      DateTime
  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model VerificationToken {
  identifier String
  token      String
  expires    DateTime

  @@id([identifier, token])
}

enum UserRole {
  USER
  ADMIN
}
```

### Environment & setup

- OS: Windows 11
- Database: PostgreSQL
- Node.js version: 20.12.2

### Prisma Version

```
Prisma Version:  5.12.1
```

**Comment by Jes015** [comment]
Interesting

**Comment by janpio** [comment]
Does this cause any problems or error, or is this just a warning?

**Comment by EddyerDevv** [comment]
It is a warn, but at the end of so many calls or restarts of the page, sometimes it does not restart and I have to restart the server, already in turbo mode because the warn does not appear.

**Comment by Hugo-Lorenzoni** [comment]
I just found out something !
I downgraded next-auth to version 4.24.3 and I have the same error message than the one I posted to abose when I try to use the Nextjs Middleware.
So the problem might come from the fact that the middleware is suppose to run on the edge or something like that and that the prisma adapter is not made for that ! 

### Middleware
```js
import { withAuth } from "next-auth/middleware";
import prisma from "./db";

export default withAuth(
  // `withAuth` augments your `Request` with the user's token.
  function middleware(req) {
    console.log(req.nextauth.token);
  },
  {
    callbacks: {
      authorized: async ({ req: { cookies } }) => {
        const sessionToken = cookies.get("next-auth.session-token")?.value;
        const user = await prisma.session
          .findUnique({ where: { sessionToken } })
          .user();
        console.log(user?.isAdmin);

        return sessionToken != null;
      },
    },
  }
);
```
### Error

> ⨯ Error: PrismaClient is not configured to run in Vercel Edge Functions or Edge Middleware. In order to run Prisma Client on edge runtime, either:
> - Use Prisma Accelerate: https://pris.ly/d/accelerate
> - Use Driver Adapters: https://pris.ly/d/driver-adapters
> 
> If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report
>     at Object.get (webpack-internal:///(middleware)/./node_modules/.prisma/client/index-browser.js:207:15)
>     at Object.authorized (webpack-internal:///(middleware)/./src/middleware.ts:18:76)
>     at handleMiddleware (webpack-internal:///(middleware)/./node_modules/next-auth/next/middleware.js:50:334)
>     at async Object.options [as handler] (webpack-internal:///(middleware)/./node_modules/next-auth/next/middleware.js:68:31)
>     at async adapter (webpack-internal:///(middleware)/./node_modules/next/dist/esm/server/web/adapter.js:179:16)
>     at async (file://C:\Code\iot-project\node_modules\next\dist\server\web\sandbox\sandbox.js:110:22)
>     at async runWithTaggedErrors (file://C:\Code\iot-project\node_modules\next\dist\server\web\sandbox\sandbox.js:107:9)
>     at async DevServer.runMiddleware (file://C:\Code\iot-project\node_modules\next\dist\server\next-server.js:1062:24)
>     at async DevServer.runMiddleware (file://C:\Code\iot-project\node_modules\next\dist\server\dev\next-dev-server.js:268:28)
>     at async NextNodeServer.handleCatchallMiddlewareRequest (file://C:\Code\iot-project\node_modules\next\dist\server\next-server.js:324:26)

**Comment by EddyerDevv** [comment]
@Hugo-Lorenzoni Well, if you are using neon serverless it works, everything and remove the prism warn from the edge functions, and that works but sometimes the Edge Functions invocations are not completed because of the async await, in the middleware it can serve as protecting routes, I I solved it by removing the middleware, since I won't need it for the moment, but I think it's a Prisma adapter error

**Comment by jkomyno** [comment]
Hi @EddyerDevv! We believe we have just fixed this issue via https://github.com/prisma/prisma/pull/24312.
You can try an integration version of `prisma` and `@prisma/client` with this fix: `5.15.0-integration-client-dynamic-wasm-imports.1`.

We expect to release this fix for everybody in `prisma@5.15.0` next week. Can you please try it and report your results here? Thanks.

**Comment by jkomyno** [comment]
> Hi @EddyerDevv! We believe we have just fixed this issue via #24312. You can try an integration version of `prisma` and `@prisma/client` with this fix: `5.15.0-integration-client-dynamic-wasm-imports.1`.
> 
> We expect to release this fix for everybody in `prisma@5.15.0` next week. Can you please try it and report your results here? Thanks.

We're closing this issue as fixed. It'll be part of version 5.15.0 of Prisma, which will be released today 🚀

**Comment by rainer85ah** [comment]
I know this is closed but perhaps can help other, I solve this issue by refactoring my auth.ts file with the help of a auth.config.ts file. The explanation and workaround is in the [official documentation](https://authjs.dev/guides/edge-compatibility) of Auth.ts for running in the edge.
